OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java

```diff
@@ -4167,6 +4170,46 @@ public void testAndroidManifestConfigChangeFlags_orientationFlag() throws Except
       assertFalse(mainHasFlag);
    }
 
+   // ####### Unit test log listener ########
+   @Test
+   public void testAddLogListener() throws Exception {
+      final String[] lastMessage = new String[1];
+      OneSignal.addLogListener(event -> lastMessage[0] = event.getEntry());
+
+      OneSignal_Log(OneSignal.LOG_LEVEL.DEBUG, "test");
+      assertEquals(lastMessage[0], "test");
+   }
+
+   @Test
+   public void testRemoveLogListener() {
+      final String[] lastMessage = new String[1];
+      OneSignalLogListener listener = event -> lastMessage[0] = event.getEntry();
+      OneSignal.addLogListener(listener);
+
+      OneSignal.removeLogListener(listener);
+      OneSignal_Log(OneSignal.LOG_LEVEL.DEBUG, "test");
+      assertNull(lastMessage[0]);
+   }
+
+   @Test
+   public void testNestedAddLogListenerDoesNotThrow() {
+      OneSignalLogListener listener2 = event -> OneSignal.addLogListener(event2 -> {});
```

Comment:
nit: does this need to be named `listener2`? It's the first listener that will add an anonymous listener in the callback.

Reply:
fixed